source/common/network/filter_manager_impl.h

```diff
@@ -136,6 +173,7 @@ class FilterManagerImpl : protected Logger::Loggable<Logger::Id::connection> {
     FilterManagerImpl& parent_;
     ReadFilterSharedPtr filter_;
     bool initialized_{};
+    bool pending_close_{false};
```

Comment:
I didn't get the usage of this bool flag.

Reply:
This is a flag to indicate if a filter is gating the close, and this flag will impact the counter and disableClose() will change its status.

Reply:
Yeah. I got it's usage. But I am not sure if it's necessary to use it. Why cannot we just increment the count when the closeDisable(true) is called?
We have a similar practice on the `read`, the `readDisable(true/false)`. I think use the similar parttern may be better?

Reply:
The key design here is to simplify the usage and lose the counter check, and we only count and check each filter's status, and the counter will only ++ when the filter status is pending. 
   *    `disableClose(true)` marks the filter as unable to close by delaying the closure process.
   *    Calling `disableClose(true)` again has no additional effect, as the closure is already marked as pending.
   *   One `disableClose(false)` can unblock the close, meaning the filter has done its job.
   
  readDisable is kind of ++ and -- mapped for each call due to the buffer scenario and some initialization scenario is very simple. But close scenario can be more complicated depending on many things which it is possible that a strict  start and end are mapped.

Reply:
> disableClose(true) marks the filter as unable to close by delaying the closure process.
Calling disableClose(true) again has no additional effect, as the closure is already marked as pending.
One disableClose(false) can unblock the close, meaning the filter has done its job.

I knew this behavior. I was thinking if there are multiple async actions block the closing, I think may be we should call the `disableClose(true)` multiple time and increment the count. And everytime once an action is done we can call the `disableClose(false)` to decrement the count.

That's say, I think the disableClose(true) and disableClose(false) should be a pair. wdyt?

Reply:
Discussed offline, and we will lose the counter check by the filter manager through only checking all filters status. We will let the filter itself to maintain the final close check.